Sources_v3/StreamChatUI/ChatChannel/MessageComposerInputAccessoryViewController.swift

```diff
@@ -262,15 +262,38 @@ open class MessageComposerInputAccessoryViewController<ExtraData: ExtraDataTypes
     
     // MARK: UITextView
     
-    func promptSuggestionIfNeeded() {
-        if textView.text == "/" {
-            showSuggestionsViewController()
+    @objc func promptSuggestionIfNeeded(for text: String) {
+        // Check if first symbol is `/` and if there are available commands in `ChatConfig`.
+        guard text.trimmingCharacters(in: .whitespacesAndNewlines).first == "/",
+            let commands = controller?.channel?.config.commands
+        else {
+            dismissSuggestionsViewController()
+            return
+        }
+        
+        // Get the command value without the `/`
+        let typedCommand = String(text.trimmingCharacters(in: .whitespacesAndNewlines).dropFirst())
+        
+        // Set all commands as hints initially
+        var commandHints: [Command] = commands
+        
+        // Filter commands when user is typing something after `/`
+        if !typedCommand.isEmpty {
+            commandHints = commands.filter { $0.name.range(of: typedCommand, options: .caseInsensitive) != nil }
+        }
+    
+        suggestionsViewController.state = .commands(commandHints)
+        suggestionsViewController.didSelectItemAt = { [weak self] index in
+            self?.state = .slashCommand(commandHints[index])
         }
+        showSuggestionsViewController()
     }
     
     func replaceTextWithSlashCommandViewIfNeeded() {
-        if textView.text == "/giphy" {
-            state = .slashCommand
+        if let command = controller?.channel?.config.commands?.first(where: { command in
+            command.name == textView.text.trimmingCharacters(in: .whitespacesAndNewlines).dropFirst()
+        }) {
+            state = .slashCommand(command)
         }
```

Comment:
🤯  Can you please structure this such that a person who didn't write this can understand? :D :D

Reply:
Yeah, agree :)
adjusted in 3af5238